ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/BlockTransactionSelector.java

```diff
@@ -241,6 +263,40 @@ private TransactionSelectionResult evaluateTransaction(
     return TransactionSelectionResult.CONTINUE;
   }
 
+  private ValidationResult<TransactionInvalidReason> validateTransaction(
+      final ProcessableBlockHeader blockHeader,
+      final Transaction transaction,
+      final WorldUpdater publicWorldStateUpdater) {
+    final MainnetTransactionValidator transactionValidator =
+        transactionProcessor.getTransactionValidator();
+    ValidationResult<TransactionInvalidReason> validationResult =
+        transactionValidator.validate(transaction, blockHeader.getBaseFee());
+    if (!validationResult.isValid()) {
+      LOG.warn(
+          "Invalid transaction: {}. Block {} Transaction {}",
+          validationResult.getErrorMessage(),
+          blockHeader.getParentHash().toHexString(),
+          transaction.getHash().toHexString());
+      return validationResult;
+    }
+
+    final Address senderAddress = transaction.getSender();
+
+    final EvmAccount sender = publicWorldStateUpdater.getOrCreate(senderAddress);
+    validationResult =
+        transactionValidator.validateForSender(
+            transaction, sender, TransactionValidationParams.processingBlock());
+    if (!validationResult.isValid()) {
+      LOG.warn(
+          "Invalid transaction: {}. Block {} Transaction {}",
+          validationResult.getErrorMessage(),
+          blockHeader.getParentHash().toHexString(),
+          transaction.getHash().toHexString());
+      return validationResult;
+    }
```

Comment:
I would do some logging when we are removing an invalid transaction from the pool in the calling method (line 261), not here.